test/gurka/test_traffic.cc

```diff
@@ -701,6 +701,94 @@ TEST(Traffic, CutGeoms) {
         EXPECT_TRUE(map.nodes["E"].ApproximatelyEqual(shapes[8]));
       }
     }
+    // Test regression when cutting
+    {
+      {
+        valhalla::baldr::TrafficSpeed ts{
+            valhalla::baldr::UNKNOWN_TRAFFIC_SPEED_RAW,
+            valhalla::baldr::UNKNOWN_TRAFFIC_SPEED_RAW,
+            valhalla::baldr::UNKNOWN_TRAFFIC_SPEED_RAW,
+            valhalla::baldr::UNKNOWN_TRAFFIC_SPEED_RAW,
+            0u,
+            0u,
+            0u,
+            0u,
+            0u,
+            0u,
+        };
+        ts.overall_speed = 36 >> 1;
+
+        ts.speed1 = 20 >> 1;
+        ts.congestion1 = baldr::MAX_CONGESTION_VAL - 1;
+        ts.breakpoint1 = 100;
+
+        ts.speed2 = 40 >> 1;
+        ts.congestion2 = 1;
+
+        // Regression is when breakpoint2 is set to 255,
+        // but speed3 is not set to UNKNOWN - we would incorrectly
+        // emit an additional 0-length slice in the API response
```

Comment:
For my understanding, the case of `speed3` or `congestion3` being set while `breakpoint2` is 255 is still an invalid state, but you're making the code more defensive to gracefully handle such an invalid state?

Reply:
Yeah - the *old* logic allowed for doing something in that case (breakpoint=255, but still emitting the third split).  The code now correctly does not make the third split if breakpoint=255, so set values in speed3/congestion3 will be properly ignored.